### ISSUE TITLE
xCreateSpell/xGetSpellInfo sometimes returns invalid data/crashes

### ISSUE THREAD

**Issue by NullCascade** [MWSE/MWSE#7]
Reported by sveng in Discord. This one is hard to pin down, and hasn't been found outside of his mod. The crash sometimes happens in looping spells, and sometimes after adding a spell.

**Comment by NullCascade** [comment]
Update:

[10:35 PM] sveng: @NullCascade Thanks, that debug DLL worked for me, whatever magic you worked!  And it's really helpful.  I found something repeatable.  Whenever targeting an item with >=6 effects (Fury, etc)  the game can not be saved or reloaded without CTD while with <=4 it can be.  Checking the log file those items have addeffect calls exceeding 8 effects.  That occurs after the item is hidden (deleting all but 1 effect) and then restored adding back >=5 effects the >=5th ones seem reported as exceeding 8.  Then I realized just adding items directly to the player (so they are immediately identified) with >1 effect there's some dis-synchronization occuring resulting in what appears to be effects not getting deleted as expected with xDeleteEffect, although a quick independent case seemed to work OK.  I need to investigate further.  None of this occurs with 0.9 but it still could be a scripting issue that is just somehow benign there.  I won't have as much time the next few days but will report back with more info as soon as I can.
